### DISCUSSION TITLE
How to prevent menu popup on right click?

### DISCUSSION THREAD

**Discussion by dpayne5** [excaliburjs/Excalibur#1813] [Q&A] [answered]
Hello,

I'm making a simple project where I'm planning on having left and right mouse clicks shoot different bullets, I am wondering how to prevent the right click popup menu from showing in the browser, what I've tried so far doesn't work: 

```
game.input.pointers.primary.on("down", function (evt) {
  evt.stopPropagation();
  if (evt.button == "Left") {
    shootLeft();
  } else {
    shootRight();
  }
});
```

I apologize in advance if this is somewhere obvious in the documentation, I haven't been able to find it so I figured I'd ask. Thanks!

**Comment by eonarheim** [comment, marked as answer]
Hi @dpayne5,

You can override the browser handler `document.oncontextmenu` and return `false` out of it to prevent the default browser behavior

https://codesandbox.io/s/context-menu-9rgk0?file=/src/index.ts

Quick example:

```
import { Actor, Color, Engine } from "excalibur";

class Game extends Engine {
  initialize() {
    const actor = new Actor({
      x: 100,
      y: 100,
      width: 50,
      height: 50,
      color: Color.Red
    });
    this.input.pointers.primary.on("move", (ev) => {
      actor.pos = ev.worldPos;
    });

    this.input.pointers.primary.on("down", (ev) => {
      console.log(ev.button);
    });

    this.add(actor);

    this.start();
  }
}

document.oncontextmenu = () => {
  return false;
};

export const game = new Game();

game.initialize();
```

**Reply by dpayne5** [reply]
Thank you very much!
